src/wifi_manager.py

```
@@ -1716,245 +2001,116 @@ def _enable_ap_mode_hostapd(self) -> Tuple[bool, str]:
-                
+                return False, f"Failed to create AP profile: {error_msg}"
+
+            # Write the NM dnsmasq-shared.d captive-portal config BEFORE bringing up
+            # the connection so NM's dnsmasq picks it up at start time.
+            # This causes every hostname DNS query from a connected device to resolve
+            # to 192.168.4.1, automatically triggering the OS captive-portal popup.
+            self._write_nm_dnsmasq_captive_conf()
+
+            logger.info("AP connection profile created, bringing it up...")
+            up_result = subprocess.run(
+                ["nmcli", "connection", "up", "LEDMatrix-Setup-AP"],
```

Ruff (0.15.12)

[error] Starting a process with a partial executable path

(S607)